packages/polar/src/lib/deploy/contract.ts

```diff
@@ -0,0 +1,94 @@
+import fs from "fs-extra";
+import path from "path";
+import { CosmWasmClient, SigningCosmWasmClient } from "secretjs";
+
+import { PolarError } from "../../internal/core/errors";
+import { ERRORS } from "../../internal/core/errors-list";
+import {
+  ARTIFACTS_DIR,
+  CONTRACTS_DIR,
+  SCHEMA_DIR
+} from "../../internal/core/project-structure";
+import { Account, PolarRuntimeEnvironment } from "../../types";
+import { getClient, getSigningClient } from "../client";
+
+export class Contract {
+  readonly contract_path: string;
+  readonly schema_path: string;
+  readonly env: PolarRuntimeEnvironment;
+  readonly client: CosmWasmClient;
+  readonly signingClient: SigningCosmWasmClient;
+
+  codeId: number;
+  contractAddress: string;
+
+  constructor (contractName: string, account: Account, env: PolarRuntimeEnvironment) {
+    this.contract_path = path.join(ARTIFACTS_DIR, "contracts", `${contractName}.wasm`);
+    this.schema_path = path.join(SCHEMA_DIR, `${contractName}.json`);
+
+    this.env = env;
+    this.client = getClient(env.network);
+    this.signingClient = getSigningClient(env.network, account);
+
+    if (!fs.existsSync(this.contract_path)) {
+      // throw new PolarError(ERRORS.ARGUMENTS.PARAM_NAME_INVALID_CASING, {
+      //   param: cLA
+      // });
+    }
+
+    if (!fs.existsSync(this.schema_path)) {
+      // TODO: log a warning that schema does not exist for this contract
+    }
+  }
+
+  async deploy (): Promise<void> {
+    const wasmFileContent: Buffer = fs.readFileSync(this.contract_path);
+
+    const uploadReceipt = await this.signingClient.upload(wasmFileContent, {});
+    const codeId: number = uploadReceipt.codeId;
+    const contractCodeHash: string =
+      await this.signingClient.restClient.getCodeHashByCodeId(codeId);
+
+    this.codeId = codeId;
```

Comment:
you can create a type `contractInfo` in types.
```
export interface ContractInfo {
       codeId: number,
       contractAddress: string
        timestamp: number
      ......

}
```
to keep contract information. and return it

Reply:
Done